src/components/cc-email-list/cc-email-list.js

```diff
@@ -77,132 +87,125 @@ export class CcEmailList extends LitElement {
 
   static get properties () {
     return {
-      addEmailForm: { type: Object },
+      addEmailFormState: { type: Object, attribute: false },
       emails: { type: Object },
     };
   }
 
-  /**
-   * @return {AddEmailFormState}
-   * @static
-   */
-  static get ADD_FORM_INIT_STATE () {
-    return {
-      state: 'idle',
-      address: {
-        value: '',
-      },
-    };
-  }
-
   constructor () {
     super();
 
-    /** @type {AddEmailFormState} The form state. */
-    this.addEmailForm = CcEmailList.ADD_FORM_INIT_STATE;
+    /** @type {AddEmailFormState} */
+    this.addEmailFormState = { type: 'idle' };
 
     /** @type {EmailListState} State of the component. */
     this.emails = { state: 'loading' };
 
-    /** @type {Ref<CcInputText>} */
+    /** @type {CcInputTextRef} */
     this._addressInputRef = createRef();
 
+    /** @type {HTMLFormElementRef} */
+    this._formRef = createRef();
+
     new LostFocusController(this, '.secondary', ({ suggestedElement }) => {
       if (suggestedElement != null) {
-        suggestedElement.querySelector('.delete-button').focus();
+        /** @type {HTMLElement} */
+        const btn = suggestedElement.querySelector('.delete-button');
+        btn.focus();
       }
       else {
-        this._focusAddressInput();
+        this._addressInputRef.value.focus();
       }
```

Comment:
suggestion: you have a `_focusBySelector` method in SSH key list. It could be moved to a utils function used in several places and return a boolean if the input was found:

```js
const wasFocused = focusBySelector(suggestedElement, '.delete-button');
if (!wasFocused) {
  this._focusAddressInput();
  this._addressInputRef.value.focus();
}
```

with the impl:

```js
/**
 * @param {Element} element
 * @param {string} selector
 */
focusBySelector (element, selector) {
  const elementToFocus = element?.querySelector(selector);
  if (typeof elementToFocus?.focus === 'function') {
    elementToFocus.focus();
    return true;
  }
  return false;
}
```

Reply:
Nice suggestion. I went for this impl:

```js
/**
 * @param {Element|ShadowRoot|Document} element
 * @param {string} selector
 * @param {() => void} [orElse]
 */
export function focusBySelector (element, selector, orElse = null) {
  /** @type {Element & { focus?: () => void}} */
  const elementToFocus = element?.querySelector(selector);
  if (typeof elementToFocus?.focus === 'function') {
    elementToFocus.focus();
  }
  else {
    orElse?.();
  }
}
```

Reply:
Interesting, this kind of idea crossed my mind but I was like "maybe too much", nice, I love it :+1:

Reply:
It's never "too much" for me :laughing: